guides/ssl-certificates.md

```diff
@@ -0,0 +1,177 @@
+the token displayed because Cloudflare will not display it again.
+
+Now that we have our Cloudflare API token, we need to configure cert-manager to
+use it. In a text editor, create a new file called **issuer.yaml** and paste the
+following:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cloudflare-api-key-secret
+  namespace: coder # Your Coder deployment namespace
+type: Opaque
+stringData:
+    api-key: "" # Your Cloudflare API token (from earlier)
+
+---
+apiVersion: cert-manager.io/v1alpha2
+kind: Issuer
+metadata:
+  name: letsencrypt
+  namespace: coder # Your Coder deployment namespace
+spec:
+  acme:
+    email: "" # Your email address (given to Let's Encrypt)
+    server: "https://acme-v02.api.letsencrypt.org/directory"
+    privateKeySecretRef:
+      name: letsencrypt-account-key
+    solvers:
+    - dns01:
+        cloudflare:
+          email: "" # Your Cloudflare email address
+          apiTokenSecretRef:
+            name: cloudflare-api-key-secret
+            key: api-key
+
+    # This section denotes which domains to use this issuer for. If you didn't
+    # limit which zones the API token had access to, you may wish to remove
+    # this section.
+    selector:
+      dnsZones:
+         # Only use this issuer for the domain example.com and its subdomains.
+         - 'example.com'
+```
+
+### ClusterIssuers
+
+cert-manager has a concept of **Issuer** (which are per-namespace) or
+**ClusterIssuer** (which are global to the entire cluster). If you plan on using
+cert-manager only for Coder, you may choose to use the **Issuer** configuration
+above. If you want to use a **ClusterIssuer** instead, you'll need to make the
+following changes:
+
+Change the namespace of the secret to **cert-manager** Change the kind of the
+**Issuer** to **ClusterIssuer** Remove the namespace of the **ClusterIssuer**
+
+For further information, see [Setting Up
+Issuers](https://docs.cert-manager.io/en/release-0.8/tasks/issuers/index.html).
+
+Read the comments and fill out the blanks. Once you're done, you can go ahead
+and apply that to your cluster using:
+
+```bash
+$ kubectl apply -f issuer.yaml
+
+secret/cloudflare-api-key-secret created
+issuer.cert-manager.io/letsencrypt created
+```
+
+## Step 3: Configure Coder to Issue and Use the Certificates
+
+If your installation uses an external egress, you'll need to configure your
+ingress to use the **coder-root-cert** and **coder-devurls-cert**.
+
+However, if you're using the default
+[ingress](https://cert-manager.io/docs/usage/ingress/) included in the helm
+chart, you can use the following helm values to configure the internal ingress
+and automatically create your certificate:
+
+```yaml
+ingress:
+  useDefault: true
+  host: "coder.example.com"
+  tls:
+    enable: true
+    hostSecretName: coder-root-cert
+    devurlsHostSecretName: coder-devurls-cert
+  additionalAnnotations:
+  - 'cert-manager.io/cluster-issuer: letsencrypt'
```

Comment:
this would be
```
  - 'cert-manager.io/issuer: letsencrypt'
```

since this tutorial covers using an "Issuer" resource.  Under the `### ClusterIssuers` section, we need to mention the user should switch to
```yaml
  - 'cert-manager.io/cluster-issuer: letsencrypt'
```

My bad, I should have made that clear.